Migrate first usable base line

Jooshua opened the pull request:
Pull Request for 

- https://github.com/Jooshua/e4CompareFramework-team2/issues/2 JavaFX Project Setup
- https://github.com/Jooshua/e4CompareFramework-team2/issues/9 Migrate ProjectExplorerKeyListener
- https://github.com/Jooshua/e4CompareFramework-team2/issues/12 Migrate SelectionListener
- https://github.com/Jooshua/e4CompareFramework-team2/issues/16 Migrate SWT Context Menu
- https://github.com/Jooshua/e4CompareFramework-team2/issues/17 Add refresh functionality to project tree

This migrates most of the ProjectExplorer to JavaFX.

Commit:
Add ProjectExplorerViewController with JavaFX

Disabled current ProjectExplorerView.


2: JavaFX Project Setup

Task-Url: https://github.com/Jooshua/e4CompareFramework/issues/2

Commit:
Add New File event handler

2: JavaFX Project Setup

Task-Url: https://github.com/Jooshua/e4CompareFramework/issues/2

timbittner committed:
Add method to convert Directory Tree to TreeItem Tree

Task #2

timbittner committed:
Add initial Tree structure on startup. Remove unnecessary wrapper.

Note that the view can not refresh yet.

Task #2

Commit:
Format ProjectExplorerViewController

2: JavaFX Project Setup

Task-Url: https://github.com/Jooshua/e4CompareFramework/issues/2

Commit:
Add setupSelectionService()

12: Migrate SelectionListener

Task-Url: https://github.com/Jooshua/e4CompareFramework/issues/12

Commit:
Register SWT context menu on the JavaFX Canvas

Deactivate JavaFX only context menu. Improve initial selection.

16: Migrate SWT Context Menu

Task-Url: https://github.com/Jooshua/e4CompareFramework/issues/16

timbittner committed:
Add Key Listeners for JavaFX Project Explorer

Task #9

valentinpurrucker committed:
Implement Refresh-Functionality and removed unused files.

Task: #17

paulo-haas commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java:
```diff
@@ -0,0 +1,243 @@
+import javafx.scene.image.WritableImage;
+
+public class ProjectExplorerViewController {
+	private static final String PROJECT_EXPLORER_VIEW_FXML = "/ui/view/ProjectExplorerView.fxml";
```
[Paths.get(...)](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Paths.html) concatenates subpaths obeying the system file separator (e.g. `/` for Linux or `\` for Windows).

Jooshua replied:
Sure

paulo-haas commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java:
```diff
@@ -0,0 +1,243 @@
+		Node imgNode = isRoot ? null : getImage(parentNode);
+
+		TreeItem<FileTreeElement> currentNode = new TreeItem<FileTreeElement>(parentNode, imgNode);
+		if (state != null) {
```
The if statements can be combined (`state != null && state.containsKey(...)`.  If state is null then the second condition won't be checked.

Jooshua replied:
Sure

paulo-haas commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java:
```diff
@@ -0,0 +1,243 @@
+					// default file icon
+					image = services.imageService.getImage(null, "icons/Explorer_View/items/file16.png");
+				}
+				;
```
Addtional semicolon.

Jooshua replied:
Removed

paulo-haas commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/ui/view/ProjectExplorerView.fxml:
```diff
@@ -0,0 +1,19 @@
+<?import javafx.scene.control.TreeView?>
+<?import javafx.scene.layout.BorderPane?>
+
+<BorderPane maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="400.0" prefWidth="600.0" xmlns="http://javafx.com/javafx/8.0.171" xmlns:fx="http://javafx.com/fxml/1" fx:controller="de.tu_bs.cs.isf.e4cf.parts.project_explorer.view.ProjectExplorerView">
```
Wird wohl korrekt sein, aber was bedeutet hier `-Infinity`? Ich habe leider in der Doku nichts gefunden. Hätten da nicht die default values ausgereicht?

paulo-haas replied:
Vielleicht könnt ihr mir ja ein Link oder so schicken :sweat_smile:

KamilRosiak replied:
Das ist damit das mit der größe des Parents wächst. Das ist quasi Integer.MAX_VALUE

KamilRosiak commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/ProjectExplorerViewController.java:
```diff
@@ -0,0 +1,243 @@
+		setupSelectionService();
+		
+		// listeners
+		scene.setOnKeyPressed(new ProjectExplorerKeyListener(context));
```
// adding key listeners to the scene

Jooshua replied:
There also will be other listeners added here

KamilRosiak commented:
- Refactor the file path strings into a FileTable/StringTable 
- complete the java doc comments for methods and classes

Commit:
Changes due to review

- Refactor the file path strings into a FileTable/StringTable
- Complete the java doc comments for methods and classes
- Minor improvements

Jooshua commented:
> 
> 
>     * Refactor the file path strings into a FileTable/StringTable
> 
>     * complete the java doc comments for methods and classes

Resolved

Blum-P commented on de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/listeners/ProjectExplorerKeyListener.java:
```diff
@@ -1,53 +1,81 @@
+	// Define key combinations here
+	KeyCombination kcRename = new KeyCodeCombination(KeyCode.R, KeyCombination.CONTROL_DOWN);	
+	
+	public ProjectExplorerKeyListener(IEclipseContext  eclipseContext) {
```
Just a little thing: double space

Blum-P reviewed (commented):
Good job. Really good code.
